### DISCUSSION TITLE
Help with installing the docker image on Synology NAS

### DISCUSSION THREAD

**Discussion by spudje** [rauversion/rauversion-phx#153] [Q&A]
Rauversion looks like what I need. A way to self-host my DJ mixes.
I tried to install the michelson/rauversion docker image on my Synology NAS but when I tried to run it it stopped with this error:
`ERROR! Config provider Config.Reader failed with:
(UndefinedFunctionError) function Mix.env/0 is undefined (module Mix is not available)
`

Anyone can help me solve the error? Do I need to provide a different value for the MIX_ENV environment variable?

Thanks!

**Comment by michelson** [comment]
Hi, this are the env vars needed:


AWS_ACCESS_KEY_ID=
AWS_S3_BUCKET=
AWS_S3_REGION=us-east-1
AWS_SECRET_ACCESS_KEY=
DATABASE_URL=postgresql://....
DISCORD_CLIENT_ID=
DISCORD_CLIENT_SECRET=
GA_ID=
GOOGLE_MAPS_KEY=
IP_INFO_API_KEY=
DOMAIN=yourdomain.com
EMAIL_ACCOUNT=email@yourdomain.com
PLATFORM_EVENTS_FEE=3
POOL_SIZE=10
SCOUT_KEY=
SCOUT_LOG_LEVEL=WARN
SCOUT_MONITOR=true
SECRET_KEY_BASE=
SENTRY_DSN=
SMTP_DOMAIN=
SMTP_PASSWORD=
SMTP_USERNAME=
STRIPE_CLIENT_ID=
STRIPE_CLIENT_SECRET=

TWITTER_CLIENT_ID=
TWITTER_CLIENT_SECRET=
VAULT_KEY=
ZOOM_CLIENT_ID=
ZOOM_CLIENT_SECRET=
MIX_ENV=prod
PEAKS_PROCESSOR=audiowaveform
OPENAI_API_KEY=
DEFAULT_LOCALE=en

**Comment by michelson** [comment]
I think that the required env vars are:

AWS_ACCESS_KEY_ID=
AWS_S3_BUCKET=
AWS_S3_REGION=us-east-1
AWS_SECRET_ACCESS_KEY=
DATABASE_URL=postgresql://....
DOMAIN=yourdomain.com

SMTP_DOMAIN=
SMTP_PASSWORD=
SMTP_USERNAME=
EMAIL_ACCOUNT=[email@yourdomain.com](mailto:email@yourdomain.com)

**Comment by spudje** [comment]
Thanks a lot!

Unfortunately this information is still too limited. I would assume the docker image contains its own PostgreSQL database. What is the DB name and username/password?
I would like to use local (external of docker container: mapped volume) storage instead of AWS/S3.

**Reply by michelson** [reply]
nom the docker need to be connected to an external pg db

**Reply by spudje** [reply]
OK, thx. And local storage, is that possible?

**Reply by michelson** [reply]
yes, it is possible but you will have to recompile the docker image yourself. in config/prod.ex you need to replace:

`config :active_storage, :service, :amazon`
to 
`config :active_storage, :service, :local`

this is not recommended way to do it tho

**Comment by spudje** [comment]
Tried it with an external PostgreSQL DB and removing the MIX_ENV variable, but it still runs into the same error:
`ERROR! Config provider Config.Reader failed with:
** (UndefinedFunctionError) function Mix.env/0 is undefined (module Mix is not available)
    Mix.env()
    (stdlib 4.0.1) erl_eval.erl:744: :erl_eval.do_apply/7
    (stdlib 4.0.1) erl_eval.erl:537: :erl_eval.expr/6
    (stdlib 4.0.1) erl_eval.erl:323: :erl_eval.expr/6
    (elixir 1.13.4) src/elixir.erl:296: :elixir.recur_eval/3
    (elixir 1.13.4) src/elixir.erl:274: :elixir.eval_forms/3
    (elixir 1.13.4) lib/code.ex:404: Code.validated_eval_string/3
    (elixir 1.13.4) lib/config.ex:260: Config.__eval__!/3

{"init terminating in do_boot",{undef,[{'Elixir.Mix',env,[],[]},{erl_eval,do_apply,7,[{file,"erl_eval.erl"},{line,744}]},{erl_eval,expr,6,[{file,"erl_eval.erl"},{line,537}]},{erl_eval,expr,6,[{file,"erl_eval.erl"},{line,323}]},{elixir,recur_eval,3,[{file,"src/elixir.erl"},{line,296}]},{elixir,eval_forms,3,[{file,"src/elixir.erl"},{line,274}]},{'Elixir.Code',validated_eval_string,3,[{file,"lib/code.ex"},{line,404}]},{'Elixir.Config','__eval__!',3,[{file,"lib/config.ex"},{line,260}]}]}}

init terminating in do_boot ({undef,[{Elixir.Mix,env,[],[]},{erl_eval,do_apply,7,[{_},{_}]},{erl_eval,expr,6,[{_},{_}]},{erl_eval,expr,6,[{_},{_}]},{elixir,recur_eval,3,[{_},{_}]},{elixir,eval_forms,3,[{_},{_}]},{Elixir.Code,validated_eval_string,3,[{_},{_}]},{Elixir.Config,__eval__!,3,[{_},{_}]}]})



Crash dump is being written to: erl_crash.dump...done`

I think I'm facing the same as described in this image: https://github.com/rauversion/rauversion-phx/issues/130

And local storage (mapped volume) would be very preferable over AWS S3 bucket.

**Comment by michelson** [comment]
Hownare you passibg the env variables? How are you stsrting the image?

**Reply by spudje** [reply]
I use portainer for that.
